plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java

```diff
@@ -65,8 +64,8 @@ protected IStatus run(IProgressMonitor monitor) {
   }
 
   private void deleteDirectories(List<File> directories) throws IOException {
-    for (int i = RECENT_DIRECTORIES_TO_KEEP; i < directories.size(); ++i) {
-      Files.walkFileTree(directories.get(i).toPath(), new DeleteAllVisitor());
+    for (int i = RECENT_DIRECTORIES_TO_KEEP; i < directories.size(); i++) {
```

Comment:
I guess that `<` should be `<=`.

Reply:
I'm wondering why the behavior changed.

Reply:
`++i` vs `i++`

Reply:
https://www.ibm.com/developerworks/library/j-noaccent/

Reply:
Oh oops!  I was wrongly thinking the pre-post increment happened during the evaluation of the comparison expression.  Please ignore (:blush:)